src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/SinkIslandCheck.java

```diff
@@ -183,7 +186,10 @@ private boolean validEdge(final AtlasObject object)
                 // Only allow car navigable highways and ignore ferries
                 && HighwayTag.isCarNavigableHighway(object) && !RouteTag.isFerry(object)
                 // Ignore any highways tagged as areas
-                && !TagPredicates.IS_AREA.test(object);
+                && !TagPredicates.IS_AREA.test(object)
+                // Ignore edges that are fully enclosed in areas with amenity values to exclude
+                && !(this.isServiceRoad((Edge) object)
+                        && this.isWithinAreasWithExcludedAmenityTags((Edge) object));
```

Comment:
nit: This is a little confusing. It's a little hard to catch the extra `()` so it's a little tricky to parse at a glance

Could this be updated to: `!this.isServiceRoad() && !this.isWithinAreaWithAmenityTags()`?

Reply:
Wouldn't it mean all service roads are not valid? 🤔 Service roads that are within the excluded areas are invalid ones.

Reply:
Ahhh, I see. I think I misread the specs the first time through. LGTM